test/src/test/java/org/neo4j/ogm/persistence/session/capability/QueryCapabilityTest.java

```diff
@@ -688,28 +696,44 @@ public void shouldLoadNodesWithUnmappedOrNoLabels() {
     @Test
     public void shouldMapCypherCollectionsToArrays() {
         Iterator<Map<String, Object>> iterator = session
-            .query("MATCH (n:User) return collect(n.name) as names", Collections.EMPTY_MAP).iterator();
+            .query("MATCH (n:User) return collect(n.name) as names", emptyMap()).iterator();
         assertThat(iterator.hasNext()).isTrue();
         Map<String, Object> row = iterator.next();
         assertThat(row.get("names").getClass().isArray()).isTrue();
         assertThat(((String[]) row.get("names")).length).isEqualTo(4);
 
         iterator = session
-            .query("MATCH (n:User {name:'Michal'}) return collect(n.name) as names", Collections.EMPTY_MAP).iterator();
+            .query("MATCH (n:User {name:'Michal'}) return collect(n.name) as names", emptyMap()).iterator();
         assertThat(iterator.hasNext()).isTrue();
         row = iterator.next();
         assertThat(row.get("names").getClass().isArray()).isTrue();
         assertThat(((String[]) row.get("names")).length).isEqualTo(1);
 
         iterator = session
-            .query("MATCH (n:User {name:'Does Not Exist'}) return collect(n.name) as names", Collections.EMPTY_MAP)
+            .query("MATCH (n:User {name:'Does Not Exist'}) return collect(n.name) as names", emptyMap())
             .iterator();
         assertThat(iterator.hasNext()).isTrue();
         row = iterator.next();
         assertThat(row.get("names").getClass().isArray()).isTrue();
         assertThat(((Object[]) row.get("names")).length).isEqualTo(0);
     }
 
+    @Test(expected = MappingException.class)
+    public void shouldThrowExceptionIfTypeMismatchesInQueryForObject() {
+        session.queryForObject(Restaurant.class, "MATCH (n:User) return count(n)", emptyMap());
```

Comment:
```suggestion
        assertThatExceptionOfType(MappingException.class)
            .isThrownBy(() -> session.queryForObject(Restaurant.class, "MATCH (n:User) return count(n)", emptyMap()))
            .withMessage("Cannot map java.lang.Long to %s. This can be caused by missing registration of %1$s.", Restaurant.class.getName());
```